addons/mrp_product_expiry/wizard/confirm_expiry.py

```diff
@@ -0,0 +1,38 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models, _
+
+
+class ConfirmExpiry(models.TransientModel):
+    _inherit = 'expiry.picking.confirmation'
+
+    produce_id = fields.Many2one('mrp.product.produce', readonly=True)
+
+    @api.depends('lot_ids')
+    def _compute_description(self):
+        if self.picking_ids:
+            super(ConfirmExpiry, self)._compute_description()
+        else:
+            if len(self.lot_ids) == 1:
+                self.description = _(
+                    "You are going to use the component %s, %s which is expired."
+                    "\nDo you confirm you want to proceed ?" % (
+                        self.lot_ids.product_id.display_name,
+                        self.lot_ids.name,
+                    )
+                )
+            else:
+                self.description = _(
+                    "You are going to use some expired components."
+                    "\nDo you confirm you want to proceed ?"
+                )
+
+    def confirm_produce(self):
+        return self.produce_id._do_produce()
```

Comment:
to check with arb. Do you not need to call again do_produce public with a context key. If multiple wizard need to be raise.